cdc/sink/codec/canal_flat.go

```diff
@@ -509,33 +531,36 @@ func canalFlatMessage2RowChangedEvent(flatMessage canalFlatMessageInterface) (*m
 	}
 
 	var err error
-	result.Columns, err = canalFlatJSONColumnMap2SinkColumns(flatMessage.getData(), flatMessage.getMySQLType())
+	result.Columns, err = canalFlatJSONColumnMap2SinkColumns(flatMessage.getData(), flatMessage.getMySQLType(), flatMessage.getJavaSQLType())
 	if err != nil {
 		return nil, err
 	}
-	result.PreColumns, err = canalFlatJSONColumnMap2SinkColumns(flatMessage.getOld(), flatMessage.getMySQLType())
+	result.PreColumns, err = canalFlatJSONColumnMap2SinkColumns(flatMessage.getOld(), flatMessage.getMySQLType(), flatMessage.getJavaSQLType())
 	if err != nil {
 		return nil, err
 	}
 
 	return result, nil
 }
 
-func canalFlatJSONColumnMap2SinkColumns(cols map[string]interface{}, mysqlType map[string]string) ([]*model.Column, error) {
+func canalFlatJSONColumnMap2SinkColumns(cols map[string]interface{}, mysqlType map[string]string, javaSQLType map[string]int32) ([]*model.Column, error) {
 	result := make([]*model.Column, 0, len(cols))
 	for name, value := range cols {
-		typeStr, ok := mysqlType[name]
+		javaType, ok := javaSQLType[name]
 		if !ok {
-			// this should not happen, else we have to check encoding for mysqlType.
+			// this should not happen, else we have to check encoding for javaSQLType.
 			return nil, cerrors.ErrCanalDecodeFailed.GenWithStack(
-				"mysql type does not found, column: %+v, mysqlType: %+v", name, mysqlType)
+				"java sql type does not found, column: %+v, mysqlType: %+v", name, javaSQLType)
 		}
-		tp := types.StrToType(typeStr)
+		mysqlTypeStr, ok := mysqlType[name]
 		if !ok {
+			// this should not happen, else we have to check encoding for mysqlType.
 			return nil, cerrors.ErrCanalDecodeFailed.GenWithStack(
-				"mysql type does not found, column: %+v, type: %+v", name, tp)
+				"mysql type does not found, column: %+v, mysqlType: %+v", name, mysqlType)
 		}
-		col := NewColumn(value, tp).ToSinkColumn(name)
+		mysqlTypeStr = strings.TrimSuffix(mysqlTypeStr, " unsigned")
```

Comment:
```suggestion
		mysqlTypeStr = strings.TrimSuffix(mysqlTypeStr, "unsigned")
```

A typo?

Could you add some comments about what `mysqlTypeStr` would look like?

Reply:
no. when encoding, we append " unsigned" to unsigned MySQL types, this just reverse it.